packages/airnode-node/src/workers/local-gateways/validation.test.ts

```diff
@@ -192,68 +198,93 @@ describe('validateAndDecodeBeacons', () => {
-  });
-
-  it('returns null if some beacons have invalid encoded parameters', () => {
-    const outdatedBeacons: Required<Beacon>[] = [
-      { ...validBeacons[0], encodedParameters: 'invalid encoded parameters' },
-      ...validBeacons.slice(1),
+      ...validBeaconsWithIds.slice(1),
     ];
 
-    expect(validateAndDecodeBeacons(outdatedBeacons)).toBeNull();
+    expect(decodeBeaconsWithData(outdatedBeacons)).toBeNull();
   });
 
   it('returns null if some beacons have invalid signature', () => {
-    const outdatedBeacons: Required<Beacon>[] = [
+    const outdatedBeacons: Required<BeaconWithIds>[] = [
       {
-        ...validBeacons[0],
+        ...validBeaconsWithIds[0],
         signedData: {
-          ...validBeacons[0].signedData,
+          ...validBeaconsWithIds[0].signedData,
           signature:
             '0x9122514d1cb4598435ea21afb7790d9daf5850b87673872b7ddf9e8df7a6afb3106a9f93557429b2b8bca4c9c57e103b93ab0de11f7bec8feeca1968189bffffff',
         },
       },
-      ...validBeacons.slice(1),
+      ...validBeaconsWithIds.slice(1),
     ];
 
-    expect(validateAndDecodeBeacons(outdatedBeacons)).toBeNull();
+    expect(decodeBeaconsWithData(outdatedBeacons)).toBeNull();
   });
 
   it('returns null if some beacons have invalid encoded value', () => {
-    const outdatedBeacons: Required<Beacon>[] = [
-      { ...validBeacons[0], signedData: { ...validBeacons[0].signedData, encodedValue: 'invalid encoded value' } },
-      ...validBeacons.slice(1),
+    const outdatedBeacons: Required<BeaconWithIds>[] = [
+      {
+        ...validBeaconsWithIds[0],
+        signedData: { ...validBeaconsWithIds[0].signedData, encodedValue: 'invalid encoded value' },
+      },
+      ...validBeaconsWithIds.slice(1),
     ];
 
-    expect(validateAndDecodeBeacons(outdatedBeacons)).toBeNull();
+    expect(decodeBeaconsWithData(outdatedBeacons)).toBeNull();
   });
 
   it('returns null if some beacons have value out of range', () => {
-    const outdatedBeacons: Required<Beacon>[] = [
+    const outdatedBeacons: Required<BeaconWithIds>[] = [
       {
-        ...validBeacons[0],
+        ...validBeaconsWithIds[0],
         signedData: {
-          ...validBeacons[0].signedData,
+          ...validBeaconsWithIds[0].signedData,
           encodedValue: '0xffffffff00000000000000000000000000000000000000000000000000000000',
         },
       },
-      ...validBeacons.slice(1),
+      ...validBeaconsWithIds.slice(1),
     ];
 
-    expect(validateAndDecodeBeacons(outdatedBeacons)).toBeNull();
+    expect(decodeBeaconsWithData(outdatedBeacons)).toBeNull();
+  });
+});
+
+describe('validateBeacons', () => {
+  it('returns beacons with beacon and template ID if they are valid', () => {
+    const beacon = validBeaconsWithIds[0];
+
+    expect(validateBeacons({ [beacon.beaconId]: validRequestBody.beacons[beacon.beaconId] })).toEqual([beacon]);
+  });
+
+  it('returns null if some beacons have invalid encoded parameters', () => {
```

Comment:
This follow the same logic as `decodeBeaconsWithData` (previously called `validateAndDecodeBeacons`) and returns `null` in case something invalid is passed. This is OK, because we assume the relay will filter out invalid beacons and not send them to the relay at all.

Also, it's reasonable to expect that invalid beacon will not be used (since it can't be used on chain anyway).